docs/design/authentication_policy.md

```diff
@@ -0,0 +1,40 @@
+# Envoy Gateway Authentication Policy API
+
+## Overview
+
+This authen policy is to declare the authentication mechanisms, to be enforce on connection and request going though Envoy Gateway. This includes the credential (X.509, JWT, etc), parameters (cipher suites, key algorithms)
+The policy is similar to [OpenAPI 3.1 security objects](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#securitySchemeObject) without the API key part, and should be easily translatable from it with some additions.
+
+## Authentication mechanisms
+This policy should support the following authentication mechanisms:
+- JWT Bearer Token
+- mutualTLS (client certificate)
+- OAuth2
+- OIDC
+- External authentication
+
+In the first phase, Envoy Gateway will implement JWT Bearer Token authentication.
+
+In general those policy translates into Envoy HTTP filters at HTTP connection manager level, and route specific settings will be applied for each route. These APIs are expressed in a Policy CRD and attached to Gateway API resources with [Policy Attachement](https://gateway-api.sigs.k8s.io/references/policy-attachment/).
+
+## JWT Bearer Token
+
+A JWT Bearer Token authentication policy will look like the following:
+
+```
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: Authentication
+metadata:
+  name: productpage
+spec:
+  type: jwt
+  jwt:
```

Comment:
if multiple issuers exist, does the user create multiple authn policies or will this policy allow the user to specify a list of issuers and associated jwksUri ?

Reply:
https://github.com/envoyproxy/gateway/pull/733 provides details. Multiple `jwtProviders` can be specified in an Authentication. The JWT auth passes if any one of the providers responds with a success. If an HTTPRoute references multiple Authentications, then at least one `jwtProviders` for each Authentication must pass for JWT auth to succeed.